openlibrary/catalog/marc/marc_base.py

```diff
@@ -30,11 +63,31 @@ def read_isbn(self, f):
             found.append(m.group(1))
         return found
 
-    def build_fields(self, want):
-        self.fields = {}
-        want = set(want)
-        for tag, line in self.read_fields(want):
-            self.fields.setdefault(tag, []).append(line)
+    def get_control(self, tag: str) -> str | None:
+        control = self.read_fields([tag])
+        _, v = next(control, (tag, None))
+        if tag == '008':
+            # Handle duplicate 008s, even though control fields are non-repeatable.
+            if others := [d for _, d in list(control) if len(d) == 40]:
+                return min(others + [v], key=lambda s: s.count(' '))
+        return v
+
+    def get_fields(self, tag: str) -> list[MarcFieldBase]:
+        return [v for _, v in self.read_fields([tag])]
```

Comment:
```suggestion
        return [v for _, v in self.read_fields([tag]) if isinstance(v, MarcFieldBase)]
```
Or change the function return type to `-> list[str, MarcFieldBase]`